vyper/parser/expr.py

```diff
@@ -1056,8 +1056,8 @@ def dict_fail(self):
 
     @staticmethod
     def struct_literals(expr, name, context):
-        o = {}
-        members = {}
+        member_subs = {}
+        member_typs = {}
```

Comment:
Shouldn't this be _OrderedDict_; the below .keys() still stays the same - so the ordering problem still remains?

Reply:
no, the ordering is insertion order for python3.6 anyways. actually the issue is exactly that the elements are inserted in insertion order, but dicts test for equality no matter the order. check out this unintuitive nonsense:
```python
>>> xs = {'a': 1, 'b': 2}
>>> ys = {'b': 2, 'a': 1}
>>> str(xs) == str(ys)
False
>>> xs == ys
True
```

so in other words, when a literal like `Foo({a: 1, b:2})` gets parsed, it has the members in a different order but still type checks as the same type as `Foo({b: 2, a: 1})`. the solution is to reorder the members either at parse-time or assignment-time, and I did the latter in this PR because we don't have good type refinement at parse-time.